src/Composer/Package/Loader/RootPackageLoader.php

```diff
@@ -93,6 +99,14 @@ public function load(array $config, string $class = 'Composer\Package\RootPackag
             }
 
             if (!isset($config['version'])) {
+                if ($this->io !== null && $config['name'] !== '__root__') {
+                    $this->io->warning(
+                        sprintf(
+                            "Composer could not detect the root package (%s) version, defaulting to "1.0.0". See https://getcomposer.org/root-version",
```

Comment:
Ah oops quotes.. Anyway with this message I'm happy I think, if you like to update the broken tests (although you could also just set a version in those tests to avoid the warning being output, it's probably cleaner).

```suggestion
                            'Composer could not detect the root package (%s) version, defaulting to "1.0.0". See https://getcomposer.org/root-version',
```